src/main/TreeQuery.cs

```diff
@@ -0,0 +1,202 @@
+
+namespace ei8.Cortex.Diary.Plugins.Tree
+{
+    public class TreeQuery
+    {
+        [QueryKey(null)]
+        public IEnumerable<Guid> Postsynaptics { get; set; }
+
+        [QueryKey(null)]
+        public Guid RegionId { get; set; }
+
+        [QueryKey("avatarurl")]
+        public string AvatarUrl { get; set; }
+
+        [QueryKey("eupm")]
+        public IEnumerable<Guid> ExpandUntilPostsynapticMirrors { get; set; }
+
+        [QueryKey("direction")]
+        public DirectionValues? DirectionValues { get; set; }
+
+        public override string ToString()
+        {
+            StringBuilder stringBuilder = new StringBuilder();
+            Type typeFromHandle = typeof(TreeQuery);
+
+            Postsynaptics?.Select(g => g.ToString()).AppendQuery(typeFromHandle.GetQueryKey("Postsynaptics"), stringBuilder);
+
+            if (RegionId != Guid.Empty)
+            {
+                if (stringBuilder.Length > 0) stringBuilder.Append("&");
+                stringBuilder.Append($"{typeFromHandle.GetQueryKey("RegionId")}={RegionId}");
+            }
+
+            if (!string.IsNullOrEmpty(AvatarUrl))
+            {
+                if (stringBuilder.Length > 0) stringBuilder.Append("&");
+                stringBuilder.Append($"{typeFromHandle.GetQueryKey("AvatarUrl")}={HttpUtility.UrlEncode(AvatarUrl)}");
+            }
+
+            ExpandUntilPostsynapticMirrors?.Select(g => g.ToString()).AppendQuery(typeFromHandle.GetQueryKey("ExpandUntilPostsynapticMirrors"), stringBuilder);
+
+            DirectionValues.AppendQuery(typeFromHandle.GetQueryKey("DirectionValues"), delegate (DirectionValues v)
+            {
+                int num = (int)v;
+                return num.ToString();
+            }, stringBuilder);
+
+            if (stringBuilder.Length > 0)
+            {
+                stringBuilder.Insert(0, '?');
+            }
+
+            return stringBuilder.ToString();
+        }
+
+        public static bool TryParse(string value, out TreeQuery result)
+        {
+            result = null;
+            bool result2 = false;
+
+            if (!string.IsNullOrWhiteSpace(value))
+            {
+                value = (value.StartsWith("?") ? value.Substring(1) : value);
+
+                // Parse query parameters with proper URL decoding
+                var parameters = ParseQueryString(value);
+
+                // Convert dictionary to string array format for existing extension methods
+                var parameterArray = ConvertParametersToArray(parameters);
+                string[] array = parameterArray.ToArray();
+
+                if (array.Length != 0)
+                {
+                    try
+                    {
+                        Type typeFromHandle = typeof(TreeQuery);
+
+                        result = new TreeQuery
+                        {
+                            Postsynaptics = ParseGuidArray(array.GetQueryArrayOrDefault(typeFromHandle.GetQueryKey("Postsynaptics"))),
```

Comment:
Use nameof(Postsynaptics) instead of a magic string. Also do for subsequent lines.

Reply:
Done